src/pages/Register.tsx

```diff
@@ -0,0 +1,160 @@
+import { createUserWithEmailAndPassword } from 'firebase/auth';
+import React, { useState } from 'react';
+import { auth, db } from '../firebase/index'
+import { setDoc, doc } from 'firebase/firestore';
+import { useNavigate } from 'react-router-dom';
+
+export function Register() {
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [fname, setFname] = useState("");
```

Comment:
utilizar nombres de variables completos, por ejemplo: firstName
```suggestion
    const [firstName, setFname] = useState("");
```